File: map-set/1.two-sum.py
#
# @lc app=leetcode id=1 lang=python3
#
# [1] Two Sum
#

# @lc code=start


class Solution:
    def twoSum(self, nums, target):
        num_map = {}

        for i, x in enumerate(nums):
            y = target - x
            if y not in num_map:
                num_map[x] = i
            else:
                return [num_map[y], i]

# @lc code=end

File: Golang/1.two-sum.go
/*
 * @lc app=leetcode id=1 lang=golang
 *
 * [1] Two Sum
 */

// @lc code=start
func twoSum(nums []int, target int) []int {
	dict := make(map[int]int)

	for idx, val := range nums {
		k := target - val
		if _, ok := dict[k]; !ok {
			dict[val] = idx
		} else {
			return []int{dict[k], idx}
		}
	}

	return nil
}

// @lc code=end

